src/types.ts

```diff
@@ -1,71 +1,160 @@
+    | 'search'
+    | 'info_open'
+    | 'info_switcher'
+    | 'info_close'
+  trade_type_name?: string
+  tab_name?: string
+  search_string?: string
+  info_switcher_mode?: 'description' | 'glossary'
+  form_source?: string
+  subform_name?: string
+  account_type?: string
 }
 type DbotTypes = {
-    action?: 'open' | 'close' | 'search' | 'delete' | 'yes' | 'no'| 'search_string' | 'choose_shortcut' | 'bot_last_modified_time' | 'delete_popup_respond' | 'push_open_button' | 'push_user_guide' | 'save_your_bot' | 'choose_your_bot' | 'delete_your_bot'
-    shortcut_name?: string
-    form_source?: string
-    search_string?: string
-    delete_popup_respond?: string
-    bot_last_modified_time?: number
-    bot_name?: string
-};
+  action?:
+    | 'open'
+    | 'close'
+    | 'search'
+    | 'delete'
+    | 'yes'
+    | 'no'
+    | 'search_string'
+    | 'choose_shortcut'
+    | 'bot_last_modified_time'
+    | 'delete_popup_respond'
+    | 'push_open_button'
+    | 'push_user_guide'
+    | 'save_your_bot'
+    | 'choose_your_bot'
+    | 'delete_your_bot'
+  shortcut_name?: string
+  form_source?: string
+  search_string?: string
+  delete_popup_respond?: string
+  bot_last_modified_time?: number
+  bot_name?: string
+}
+
+type IndicatorsTypesFormAction = {
+  action:
+    | 'open'
+    | 'close'
+    | 'add_active'
+    | 'clean_all_active'
+    | 'delete_active'
+    | 'edit_active'
+    | 'search'
+    | 'info_open'
+    | 'info_close'
+  form_name?: string
+  indicator_type_name?: string
+  indicators_category_name?: string
+  search_string?: string
+  subform_name?: string
+  account_type: string
+  device_type: string
+}
+
+type MarketTypesFormAction = {
+  action:
+    | 'open'
+    | 'close'
+    | 'choose_market_type'
+    | 'search'
+    | 'info_redirect'
+    | 'add_to_favorites'
+    | 'delete_from_favorites'
+  form_name: string
+  market_type_name: string
+  search_string?: string
+  tab_market_name?: string
+  account_type: string
+  device_type: string
+}
 
 export type TEvents = {
-    ce_partner_account_signup_form: PartnerAccountSignupForm
-    ce_virtual_signup_form: VirtualSignupForm
-    ce_real_account_signup_form: RealAccountSignupForm
-    ce_virtual_signup_email_confirmation: VirtualSignupEmailConfirmation
-    ce_trade_types_form: TradeTypesForm
-    ce_bot_dashboard_form: DbotTypes
+  ce_partner_account_signup_form: PartnerAccountSignupForm
+  ce_virtual_signup_form: VirtualSignupForm
+  ce_real_account_signup_form: RealAccountSignupForm
+  ce_virtual_signup_email_confirmation: VirtualSignupEmailConfirmation
+  ce_trade_types_form: TradeTypesForm
+  ce_bot_dashboard_form: DbotTypes
+  ce_indicators_types_form: IndicatorsTypesFormAction
+  ce_market_types_form: MarketTypesFormAction
 }
 
 export type TCoreTrackData = {
-    language: string
-    account_type:  'CR' | 'VR' | 'MF' | 'VRTC'
-    device_type: 'mobile' | 'desktop'
-}
\ No newline at end of file
+  language: string
+  account_type: 'CR' | 'VR' | 'MF' | 'VRTC'
```

Comment:
btw can you update this types? I'm not sure that I add all of types

Reply:
this, in deriv-app is represented as `string` only

Reply:
you think better to keep it as string? so let's change it then 🙏